Classes/common/CDTIncrementalStore.m

```diff
@@ -2208,6 +2221,75 @@ - (id)executeSaveRequest:(NSSaveChangesRequest *)saveRequest
     return @[];
 }
 
+- (id)executeBatchUpdateRequest:(NSBatchUpdateRequest *)updateRequest
+                    withContext:(NSManagedObjectContext *)context
+                          error:(NSError **)error
+{
+    NSError *err = nil;
+
+    NSFetchRequest *fetchRequest = [[NSFetchRequest alloc] init];
+    fetchRequest.entity = updateRequest.entity;
+    fetchRequest.predicate = updateRequest.predicate;
+
+    NSDictionary *query = [self queryForFetchRequest:fetchRequest error:&err];
+
+    if (!query) {
+        if (error) *error = err;
+        return nil;
+    }
+
+    err = nil;
+    CDTQueryResult *hits = [self.indexManager queryWithDictionary:query options:nil error:&err];
+    // hits == nil is valid, get rid of this once tested
+    if (!hits && err) {
+        if (error) *error = err;
+        return nil;
+    }
+
+    // Note: This dictionary is *not* the same dictionary specified in the original request.
+    // The keys have all been transformed from attribute names into NSAttributeDescriptions
+    NSDictionary *updates = updateRequest.propertiesToUpdate;
+
+    NSEntityDescription *entity = [updateRequest entity];
+
+    NSMutableArray *results = [NSMutableArray array];
+    for (CDTDocumentRevision *rev in hits) {
+        NSManagedObjectID *moid = [self newObjectIDForEntity:entity referenceObject:rev.docId];
+        NSManagedObject *mo = [context objectWithID:moid];
```

Comment:
http://matthewmorey.com/core-data-batch-updates/ says:
> When calling executeRequest:error: it bypasses the managed object context and updates the persistent store directly. Because none of the changes are reflected in the context, validation rules are not performed which means you need to be careful as you can create conflicts.

So I don't think you should be creating a `NSManagedObject` but tapping the document directly.
This would require splitting out `-updateManagedObject:error:`.